softlayer/resource_softlayer_virtual_guest.go

```diff
@@ -982,14 +979,20 @@ func WaitForVirtualGuestAvailable(d *schema.ResourceData, meta interface{}) (int
 				return false, "retry", nil
 			}
 
+			// Check active transactions
 			log.Println("Checking active transactions.")
 			if result.ActiveTransaction != nil {
 				return result, "provisioning", nil
 			}
 
 			// Check Primary IP address availability.
-			log.Println("Checking Primary IP address.")
-			if sl.Get(result, ipAddress) == "" {
+			log.Println("Checking primary backend IP address.")
+			if result.PrimaryBackendIpAddress == nil {
```

Comment:
`if sl.Get(result.PrimaryBackendIpAddress, "") == "" {`

I think that would have worked, and it also checks for a blank... although I don't know if that is possible in the API response.